optax/_src/transform.py

```diff
@@ -605,3 +605,28 @@ def update_fn(updates, state, params=None):
     return updates, ApplyEvery(count=count_inc % k, grad_acc=grad_acc)
 
   return GradientTransformation(init_fn, update_fn)
+
+
+class CentralState(OptState):
+  """The `centralize` transformation is stateless."""
+
+
+def centralize() -> GradientTransformation:
+  """Centralize gradients.
+
+  References:
+    [Yong et al, 2020](https://arxiv.org/abs/2004.01461)
+
+  Returns:
+    An (init_fn, update_fn) tuple.
+  """
+
+  def init_fn(_):
+    return CentralState()
+
+  def update_fn(updates, state, params=None):
+    del params
+    updates = [up - up.mean(tuple(range(1, len(up.shape))), keepdims=True) for up in updates]
```

Comment:
please use `g` instead

Reply:
updates can be an arbitrary data structure (e.g. a nested list) so just looping with a `for` won't be sufficient,
please try:

```
updates = jax.tree_map(
    lambda g: g - g.mean(tuple(range(1, len(g.shape))), keepdims=True), updates)
```

Reply:
Thanks for the contribution!

I sent you a suggestion on how to use `tree_map`.

Could you also expose the new function in the top level `__init__.py`?
(importing it there, and adding it to the public API defined by `__all__`).

Finally would be great to add a custom test in `transform_test.py`,
stepping the transform once on some dummy inputs,
and then comparing the outputs to some hardcoded values,

You could compute the expected targets either manually,
or even better, using the authors's implementation of the optimiser (if you can find one on github).